internal/store/endpointslice.go

```diff
@@ -74,6 +74,56 @@ func endpointSliceMetricFamilies(allowAnnotationsList, allowLabelsList []string)
 				}
 			}),
 		),
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_endpointslice_endpoints_hints",
+			"Topology routing hints attached to endpoints",
+			metric.Gauge,
+			basemetrics.ALPHA,
+			"",
+			wrapEndpointSliceFunc(func(e *discoveryv1.EndpointSlice) *metric.Family {
+				m := []*metric.Metric{}
+				for _, ep := range e.Endpoints {
+					// Hint is populated when the endpoint is configured to be zone aware and preferentially route requests to its local zone.
+					// If there is no hint, skip this metric
+					if ep.Hints != nil && len(ep.Hints.ForZones) > 0 {
+						var (
+							labelKeys,
+							labelValues []string
+						)
+
+						if ep.Hostname != nil {
+							labelKeys = append(labelKeys, "hostname")
+							labelValues = append(labelValues, *ep.Hostname)
+						}
+
+						// Did you know that k8s will "assign" endpoints from a zone
+						// This field is useful for debugging weird network routing issues
+						if ep.Zone != nil {
+							labelKeys = append(labelKeys, "endpoint_zone")
+							labelValues = append(labelValues, *ep.Zone)
+						}
```

Comment:
the zone is already part of `kube_endpointslice_endpoints`, I don't think we should duplicate this info

Reply:
removed

Reply:
removed